src/js/sonifier.js

```diff
@@ -0,0 +1,402 @@
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return noteDurationConfig.divisorDuration;
+            } else {
+                return noteDurationConfig.remainderDuration;
+            }
+        });
+        return durations;
+    };
+
+
+    floe.chartAuthoring.sonifier.getSonificationNoteValues = function(units, unitDivisor, noteValueConfig) {
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return noteValueConfig.divisorValue;
+            } else {
+                return noteValueConfig.remainderValue;
+            }
+        });
+        return durations;
+    };
+
+    floe.chartAuthoring.sonifier.getSonificationEnvelopeDurations = function(units, unitDivisor, envelopeDurationConfig) {
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return [envelopeDurationConfig.divisorDuration,envelopeDurationConfig.divisorSilence];
+            } else {
+                return [envelopeDurationConfig.remainderDuration,envelopeDurationConfig.remainderSilence];
+            }
+        });
+
+        var durationsJoined = [];
+
+        fluid.each(durations, function(duration) {
+            durationsJoined = durationsJoined.concat(duration);
+        });
+
+        return durationsJoined;
+    };
+
+    floe.chartAuthoring.sonifier.getSonificationEnvelopeValues = function(envelopeDurations, envelopeDurationConfig, envelopeValuesConfig) {
+        var envelopeValues = fluid.transform(envelopeDurations, function(duration) {
+            if(duration === envelopeDurationConfig.divisorDuration || duration === envelopeDurationConfig.remainderDuration) {
+                return envelopeValuesConfig.openValue;
+            } else {
+                return envelopeValuesConfig.closedValue;
+            }
+        });
+        return envelopeValues;
+    };
+
+    // Get the synthOptions with a zoom factor applied to the durations
+    floe.chartAuthoring.sonifier.applyZoomToDurationConfig = function(durationConfig, zoom) {
+
+        var zoomedNoteDurationConfig = fluid.transform(durationConfig, function (duration) {
+            return duration * zoom;
+        });
+
+        return zoomedNoteDurationConfig;
+    };
+
+    // Kicks off a sonification play. Specifically it:
+    // - creates the needed Flocking synth
+    // - starts the Flocking environment
+    // - calls the initial playDataset function that's used recursively to
+    // execute a sonification
+
+    floe.chartAuthoring.sonifier.startSonification = function(that) {
+        if(!that.model.isPlaying) {
+            // Fire the start event
+
+            that.events.onSonificationStarted.fire();
+
+            flockingEnvironment.start();
+
+            fluid.defaults("floe.chartAuthoring.dataPianoBand", {
```

Comment:
I think this is an inadequately refactored legacy of the "get something that works quickly" code from last November. I'll see about reimplementing this so that we only define it once.